### PR TITLE
Add 5 new functions 

### PR DESCRIPTION
**Pull request by dmachadol**
get_meas_dates(cont, start, end):
Get data between two dates.

Rol and athorizations functions:
get_rols:
Get list of available rols in account profile.

set_rol(visId):
Set active rol [visId]

get_authorized():
Get list of authorized clients

get_authorized_cups(id)
Get list of cups from authorized client [id]

### CODE REVIEW

File: src/EdistribucionAPI/Edistribucion.py [hunk 1]
```diff
@@ -514,3 +514,58 @@ def get_meas_interval(self, cont, startDate, endDate):
...
+            "WP_Measure_Detail_Filter_Advanced_v3",
+            {"contId":""+cont['Id']+"","type":"4","startDate":""+start+"","endDate":""+end+""}
+        )
+        r = self.__run_action_command(action)
+        if 'data' in r.keys():
+            if 'lstData' in r['data'].keys():
+                return r['data']['lstData']
+        return []
+
+    def get_rols(self):
```
Comment (polhenarejos) [inline src/EdistribucionAPI/Edistribucion.py:531]: Seems duplicated of `get_login_info()`.

**Review by polhenarejos — COMMENTED** [review]
In general, it seems all these functions are already implemented somehow. Instead of returning specific fields from the JSON struct, actual functions returns full struct which can be particularized for each specific purpose.

Please check it.

**Comment by trocotronic** [comment]
@polhenarejos' comments are correct. Please fix it and send a new PR. Thanks.

**Comment by dmachadol** [comment]
Yes, are implemented to help calls. But Set_rol are new function that not have EdistribtionAPI.pyEl 5 nov. 2023, a les 1:03, Pol Henarejos ***@***.***> va escriure:﻿
@polhenarejos commented on this pull request.

In general, it seems all these functions are already implemented somehow. Instead of returning specific fields from the JSON struct, actual functions returns full struct which can be particularized for each specific purpose.
Please check it.

In src/EdistribucionAPI/Edistribucion.py:
> @@ -514,3 +514,58 @@ def get_meas_interval(self, cont, startDate, endDate):
 
         r = self.__run_action_command(action)
         return r['data']['lstData']
+
+    def  get_meas_dates(self, cont, start, end):

Seems a duplicated get_meas_interval(). Is there any difference?



In src/EdistribucionAPI/Edistribucion.py:
> +    def set_rol(self, visId):
+        action = EdistribucionMessageAction(
+            595,
+            "WP_ProfileMenu_CTRL/ACTION$saveRole",
+            "WP_ProfileMenu",
+            {"visId":""+visId+""}
+        )
+        r = self.__run_action_command(action)
+        return r
+
+    def get_authorized(self):
+        action = EdistribucionMessageAction(
+            390,
+            "WP_Monitor_CTRL/ACTION$getLoginInfo",
+            "WP_Monitor",
+            {"serviceNumber":"S001"}

Seems duplicated of get_login_info().



In src/EdistribucionAPI/Edistribucion.py:
> +            {"visId":""+visId+""}
+        )
+        r = self.__run_action_command(action)
+        return r
+
+    def get_authorized(self):
+        action = EdistribucionMessageAction(
+            390,
+            "WP_Monitor_CTRL/ACTION$getLoginInfo",
+            "WP_Monitor",
+            {"serviceNumber":"S001"}
+        )
+        r = self.__run_action_command(action)
+        return r['authList']
+
+    def get_authorized_cups(self, id):

It is duplicated from get_list_cups().



In src/EdistribucionAPI/Edistribucion.py:
> @@ -514,3 +514,58 @@ def get_meas_interval(self, cont, startDate, endDate):
 
         r = self.__run_action_command(action)
         return r['data']['lstData']
+
+    def  get_meas_dates(self, cont, start, end):
+        action = EdistribucionMessageAction(
+            965,
+            "WP_Measure_v3_CTRL/ACTION$getChartPointsByRange",
+            "WP_Measure_Detail_Filter_Advanced_v3",
+            {"contId":""+cont['Id']+"","type":"4","startDate":""+start+"","endDate":""+end+""}
+        )
+        r = self.__run_action_command(action)
+        if 'data' in r.keys():
+            if 'lstData' in r['data'].keys():
+                return r['data']['lstData']
+        return []
+
+    def get_rols(self):

Seems duplicated of get_login_info().

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>
[
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/trocotronic/edistribucion/pull/31#pullrequestreview-1713888409",
"url": "https://github.com/trocotronic/edistribucion/pull/31#pullrequestreview-1713888409",
"name": "View Pull Request"
},
"description": "View this Pull Request on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Pull request closed without merging**